src/otx/v2/adapters/torch/mmengine/mmdet/modules/datasets/pipelines/load_pipelines.py

```diff
@@ -102,32 +103,42 @@ def _get_annotation_mmdet_format(
             dict: annotation information dict in mmdet format
         """
         # load annotations for item
-        gt_bboxes = []
-        gt_labels = []
-        gt_polygons: list | None = []
-        gt_ann_ids = []
+        gt_bboxes: list[list[float]] = []
+        gt_labels: list[int] = []
+        gt_polygons: list[list[np.array]] = []
+        gt_ann_ids: list[tuple[str, int]] = []
 
-        width, height = dataset_item.media.data.shape[:2]
+        image_height, image_width = dataset_item.media.data.shape[:2]
```

Comment:
This may require decoding an image, so that it is good to call this when it is really required.

Reply:
Polygon mask of mmdet requires entire image's width and height as inputs. Therefore this is unremovable.
What do you think?

Reply:
So I suggested move this to L138.

Reply:
Done by [afdfb53](https://github.com/openvinotoolkit/training_extensions/pull/2644/commits/afdfb53cf8b2e48227f7492eabec2bcbf5ec9467)